src/Server.php

```diff
@@ -10,25 +10,35 @@ class Server extends EventEmitter implements ServerInterface
 {
     public $master;
     private $loop;
+    private $isSecure = false;
 
     public function __construct(LoopInterface $loop)
     {
         $this->loop = $loop;
     }
 
-    public function listen($port, $host = '127.0.0.1')
+    public function listen($port, $host = '127.0.0.1', $streamContext = array())
     {
         if (strpos($host, ':') !== false) {
             // enclose IPv6 addresses in square brackets before appending port
             $host = '[' . $host . ']';
         }
+        if (isset($streamContext['ssl'])) {
+            $this->isSecure = true;
+        }
 
-        $this->master = @stream_socket_server("tcp://$host:$port", $errno, $errstr);
+        $this->master = @stream_socket_server(
+            "tcp://$host:$port",
+            $errno,
+            $errstr,
+            STREAM_SERVER_BIND | STREAM_SERVER_LISTEN,
+            stream_context_create($streamContext)
+        );
         if (false === $this->master) {
             $message = "Could not bind to tcp://$host:$port: $errstr";
             throw new ConnectionException($message, $errno);
         }
-        stream_set_blocking($this->master, 0);
+        stream_set_blocking($this->master, (int) $this->isSecure);
```

Comment:
I would love to know if anybody has managed to solve the requirement for this master socket to block for secure connections.